src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTextUI.java

```diff
@@ -647,6 +647,21 @@ ActionMap createActionMap() {
                 TransferHandler.getCopyAction());
         map.put(TransferHandler.getPasteAction().getValue(Action.NAME),
                 TransferHandler.getPasteAction());
+
+        if (getComponent() instanceof JPasswordField) {
+            // Create the action map for Password Field.  This map provides
+            // same actions for double mouse click and
+            // and for triple mouse click (see bugs 4231444, 8354646).
+
+            if (map.get(DefaultEditorKit.selectWordAction) != null) {
+                Action a = map.get(DefaultEditorKit.selectLineAction);
+                if (a != null) {
+                    map.remove(DefaultEditorKit.selectWordAction);
+                    map.put(DefaultEditorKit.selectWordAction, a);
+                }
+            }
+        }
+
         return map;
```

Comment:
This PR could produce weird results if anyone tried applying a BasicPasswordFieldUI (or SynthPasswordFieldUI or AquaTextPasswordFieldUI ) to a JTextField that is NOT a JPasswordField.

Do we care about that use case?

(If we do: then we could simply copy and paste the same known solution into AquaTextPasswordFieldUI ? That would make this PR less invasive, but it would involve more copied and pasted code.)

Or instead of relying on `instanceof JPasswordField` would rely on `"PasswordField".equals(getPropertyPrefix()`. If anyone has any strong feelings.